solr/core/src/test/org/apache/solr/handler/TestSnapshotCoreBackup.java

```diff
@@ -452,7 +452,8 @@ private static void simpleBackupCheck(
           Files.exists(backup.resolve(expectedSegmentsFileName)));
     }
     try (Directory dir = FSDirectory.open(backup)) {
-      TestUtil.checkIndex(dir, true, true, true, null);
+      // Lucene 10 changed CheckIndex levels from 0-3 to 1-3, so we use 1 (minimum level)
+      TestUtil.checkIndex(dir, 1, true, true, null);
```

Comment:
minor: `CheckIndex.Level.*` enum could be used here -- I can include that in #3495